Server/Internal/ServerCallHandler.cs

```diff
@@ -83,24 +90,29 @@ public async Task HandleCallAsync(HttpContext httpContext)
             var service = activator.Create();
 
             // Select procedure using reflection
-            var handlerMethod = typeof(TImplementation).GetMethod(method.Name);
+            var handlerMethod = typeof(TImplementation).GetMethod(_name);
+
+            // Create the stream writer
+            var serializerDelegate = Delegate.CreateDelegate(typeof(Func<IMessage, byte[]>), typeof(ServerStreamingServerCallHandler<TImplementation>), nameof(DefaultSerializer));
+            var writerType = typeof(HttpContextStreamWriter<>).MakeGenericType(_output.ClrType);
```

Comment:
Calling MakeGenericType here will add reflection overhead for every request.

Instead I think you should revert the handler implementations back to how they were, i.e. `ServerStreamingServerCallHandler<TRequest, TResponse, TImplementation>`, and build the generic type in the route builder: `typeof(ServerStreamingServerCallHandler<,,>).MakeGenericType(...)`

The extra MakeGenericType reflection happens on startup.

Reply:
Good idea